projects/netgrif-components-core/src/lib/forms/change-password/abstract-change-password.component.ts

```diff
@@ -0,0 +1,108 @@
+import {passwordValidator} from '../models/password.validator';
+import {LoggerService} from '../../logger/services/logger.service';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {MessageResource} from '../../resources/interface/message-resource';
+import {TranslateService} from '@ngx-translate/core';
+import {Component, EventEmitter, Inject, OnDestroy, Optional, Output} from '@angular/core';
+import {FormSubmitEvent, HasForm} from "../has-form";
+import {LoadingEmitter} from "../../utility/loading-emitter";
+import {take} from "rxjs/operators";
+import {UserChangePasswordRequest} from "../../authentication/profile/models/user-change-password-request";
+import {ProfileService} from "../../authentication/profile/services/profile.service";
+import {UserService} from "../../user/services/user.service";
+import {encodeBase64} from '../../utility/base64';
+import {NAE_DEFAULT_MIN_PASSWORD_LENGTH, NAE_MIN_PASSWORD_LENGTH} from "../min-password-length-token";
+
+@Component({
+    selector: 'ncc-abstract-change-password',
+    template: ''
+})
+export abstract class AbstractChangePasswordComponent implements HasForm, OnDestroy {
+
+    public rootFormGroup: FormGroup;
+    public hideOldPassword: boolean;
+
+    public hidePassword: boolean;
+    public hideRepeatPassword: boolean;
+
+    @Output() public formSubmit: EventEmitter<FormSubmitEvent>;
+    @Output() public changePassword: EventEmitter<MessageResource>;
+
+    public loadingSubmit: LoadingEmitter;
+
+    protected constructor(protected formBuilder: FormBuilder,
+                          protected profileService: ProfileService,
+                          protected user: UserService,
+                          protected _log: LoggerService,
+                          protected _translate: TranslateService,
+                          @Optional() @Inject(NAE_MIN_PASSWORD_LENGTH) protected minPasswordLength: number | null) {
```

Comment:
Please change the type to `number | undefined` as a number cannot have a value of null.

Reply:
Done